### DISCUSSION TITLE
Stale data in Dev mode

### DISCUSSION THREAD

**Discussion by Ipicon** [vercel/next.js#58980] [App Router]
Everything in Dev mode takes 2 reloads to show and I don't know why.
I have a page that shows a user unread messages, and after reading one of them it shouldn't show the read messages anymore on refresh, but in dev mode one refresh always give back old data (even thought the server sends new one), also when changing users it still shows the old user's data until I do a full refresh.

I'm sure this is a dynamic route, because I'm using next-auth with cookies to check whether the user is logged in or not.

it's probably has something to do with caching, but in production this doesn't happen and I don't understand why.

**Comment by icyJoseph** [comment]
Are you using a Service Worker? Maybe `next-pwa`?

**Reply by Ipicon** [reply]
no, nothing like that, I'm using trpc to communicate with the server. 
I've tried revalidating the relevant queries on mutation and the issue still persists.
if trpc affected the cache, then it wouldn't explain why this works different in prod.

**Comment by Ipicon** [comment]
okay, I've added some logging and for some reason I don't understand even how, when I refresh it renders the previous "cached" Server Component and tonly after that it calculates the new server component which will be available in the next refresh.

for some context:
I have a pretty an easy GET request which print "server" with the data to the console,
and inside my server component I have something like so:
`
export default async function Home() {
    console.log("getting data");
    const data = await getData();
   console.log("client", data);
}
`
and the logs are in the following order:
getting data
client old data
server new data

**Reply by Ipicon** [reply]
for some reason it doesn't wait for the new data to come and renders an older version of the component.

**Comment by Ipicon** [comment]
ok, false alarm it just seems that trpc caches the requst, and I cant invalidate it because invalidate param only works right now for client calls.

**Comment by Ipicon** [comment]
https://discord-questions.trpc.io/m/1150707553068650579
